FlowCrypt/src/main/java/com/flowcrypt/email/api/email/javamail/AttachmentInfoDataSource.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
+ * Contributors: DenBond7
+ */
+
+package com.flowcrypt.email.api.email.javamail
+
+import android.content.Context
+import android.net.Uri
+import android.text.TextUtils
+import com.flowcrypt.email.Constants
+import com.flowcrypt.email.api.email.model.AttachmentInfo
+import java.io.BufferedInputStream
+import java.io.ByteArrayInputStream
+import java.io.InputStream
+import java.io.OutputStream
+import javax.activation.DataSource
+
+/**
+ * The [DataSource] realization for a file which received from [Uri]
+ *
+ * @author Denis Bondarenko
+ *         Date: 12/29/21
+ *         Time: 5:27 PM
+ *         E-mail: DenBond7@gmail.com
+ */
+open class AttachmentInfoDataSource(private val context: Context, val att: AttachmentInfo) :
+  DataSource {
+
+  override fun getInputStream(): InputStream? {
+    val inputStream: InputStream? = if (att.uri == null) {
```

Comment:
I can apply only a half of changes.

`context.contentResolver.openInputStream(it)` can return `null`

Reply:
@DenBond7 The additional point here is that we also don't need BufferedInputStream over ByteArrayInputStream, because ByteArrayInputStream already actually works over the buffer and supports marking. So then could be
```kotlin
return if (att.uri == null) {
  att.rawData?.inputStream()
} else {
  att.uri?.let {
    val stream = context.contentResolver.openInputStream(it)
    return if (stream != null) BufferedInputStream(stream) else null
  }
}
```

Reply:
Updated